deviceModeInit.js

```diff
@@ -95,6 +96,117 @@ var rudderTracking = (function () {
+        }
+        const percentVisible = pixelsVisible / height;
+
+        if (percentVisible > 0.8) {
+          return true;
+        }
+      }
+
+      return false;
+    };
+    function getAllProductsOnPage() {
+      const allAnchorTags = document.getElementsByTagName("a");
+      const productUrlRegex = new RegExp(
+        `^(?!\/\/cdn)[-\.:\/,a-z,A-Z,0-9]*\/products\/`
+      );
+      const allAnchorTagsWithProducts = Array.prototype.slice
+        .call(allAnchorTags)
+        .filter((anchorTag) => {
+          return productUrlRegex.test(anchorTag.href);
+        });
+      let allProductsOnPageWithImg = allAnchorTagsWithProducts.filter(
+        (anchorTag) => {
+          anchorTag.querySelector("img");
+        }
+      );
+      if (allProductsOnPageWithImg.length === 0) {
+        allProductsOnPageWithImg = allAnchorTagsWithProducts.filter(
+          (anchorTag) => {
+            const parentNode = anchorTag.parentNode.parentNode.parentNode;
+            return (
+              parentNode.nextElementSibling?.querySelector("img") ||
+              parentNode.previousElementSibling?.querySelector("img")
+            );
+          }
+        );
+      }
+      return allProductsOnPageWithImg;
+    }
+    function callProductListViewedEvent() {
+      const elementsToSend = [];
+      const productsToSend = [];
+      productsOnThePage.forEach((element, index) => {
+        const isProductVisible = productIsVisible(window, element);
+        const isProductAlreadyConsidered = productConsidered.includes(
+          element.href
+        );
+        if (isProductVisible && !isProductAlreadyConsidered) {
+          elementsToSend.push(element);
+          productConsidered.push(element.href);
+        }
+      });
+      if (elementsToSend.length > 0) {
+        elementsToSend.forEach(async (product) => {
+          const handle = product.href.match(
+            /(\/products\/)((\w|-)*)(\?|\$?)/
+          )[2];
+          await rs$
+            .get(`/products/${handle}.json`, undefined, undefined, "JSON")
+            .then((json) => {
+              console.log("Captured through rudderstack script:", json);
+              productsToSend.push(json);
+            })
+            .catch((ex) => {
+              console.debug("Littledata unable to fetch", handle, ex);
+            });
+        });
+        window.setTimeout(() => {
+          if (productsToSend.length > 0) {
+            rudderanalytics.track("Product List Viewed", {
+              products: productsToSend,
+            });
+          }
+        }, 2500);
+      }
+    }
+    productListViews();
+    function productListViews() {
+      productsOnThePage = getAllProductsOnPage();
+      let waitForScroll = window.setTimeout(() => {
+        callProductListViewedEvent();
+      }, 200);
+      document.addEventListener("scroll", () => {
+        //assumes that people need 200ms after scrolling stops to register an impression
```

Comment:
200ms would be 0.2ms. I think it should be more, maybe 1000ms (1s) or even more.
Taking a case of 3-5 products in a list, 0.2s for 3-5 products is not enough time for a shopping user to read through all the product names.

Reply:
Its fine these are basic fine tunes we can argue as long as they are seeing for .2 sec. we register these as Product Viewes

Reply:
Sure.